kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -128,10 +140,25 @@ public void produce(
             serializedValue,
             request.getTimestamp().orElse(Instant.now()));
 
-    return produceResult.thenApply(
-        result ->
-            toProduceResponse(
-                clusterId, topicName, keyFormat, keySchema, valueFormat, valueSchema, result));
+    return produceResult
+        .handle(
+            (result, ex) -> {
+              if (ex != null) {
+                recordErrorMetrics();
+                throw new CompletionException(ex);
+              }
+              return result;
+            })
+        .thenApply(
+            result -> {
+              ProduceResponse response =
+                  toProduceResponse(
+                      clusterId, topicName, keyFormat, keySchema, valueFormat, valueSchema, result);
+              long latency =
+                  Duration.between(requestInstant, result.getCompletionTimestamp()).toMillis();
+              recordResponseMetrics(latency);
+              return response;
```

Comment:
nit: What do you think about also recording latency for error requests? It feels like Coordinated Omission otherwise.

Reply:
fixed